Lombiq.HelpfulLibraries.OrchardCore/Contents/ContentTypeDefinitionBuilderExtensions.cs

```diff
@@ -7,24 +7,25 @@ public static class ContentTypeDefinitionBuilderExtensions
 {
     /// <summary>
     /// Merges all <see langword="bool"/> content type settings; the same ones you have checkboxes for in the admin
-    /// dashboard's content type editor in the same order. All values have a default false value, so you can set any
-    /// combination using parameter names in a way that signals intentionality (ie. if you explicitly want to deny an
-    /// ability or you just don't care) which is useful clarification for later updates in the migration.
+    /// dashboard's content type editor in the same order. All values have a default <see langword="null"/> value, which
+    /// means they don't get altered. If you use this in itself they are left as the default <see langword="false"/>. So
+    /// you can set any combination using parameter names in a way that signals intentionality (ie. if you explicitly

```

Comment:
Pedantic mode ON: it should be [`i.e.`](https://www.oxfordlearnersdictionaries.com/definition/english/i-e):

```suggestion
    /// you can set any combination using parameter names in a way that signals intentionality (i.e. if you explicitly
```

I even had to look up intentionality because I've never seen that and my FF browser dictionary doesn't know it either ;-)

Reply:
You wouldn't. Browser dictionaries can't help with any a.b.b.r. becuase they (very reasonably) think the dot is the end of a sentence.

Also I'm just starting to wonder if "i.e." is even needed most of the time, unless your goal is specifically to sound pretentious. :) (so probably should be avoided in technical documents) In most cases it is surrounded with parentheses which already serve the same semantic purpose more effectively. Also it's just unnecessary Latin for "that is". What do you think?

Reply:
I do find `i.e.` useful at times and use it regularly, although maybe not often.

> unless your goal is specifically to sound pretentious. :)

Well, I don't feel it that way ;-)

I agree, though, that using `i.e.` **and** parentheses is pushing it.

In the current example I would drop the parentheses, though, not the `i.e.`, because a clarification of the _intentionality_ seems useful here. What I don't really understand though is the part `or you just don't care`. If you don't care, why would you specify a parameter for it?

Reply:
> I agree, though, that using i.e. and parentheses is pushing it.

I wrote that exactly because almost every time I've seen i.e. it's inside parentheses so it's redundant. If you use it after comma, like in the example in your original link that's legitimate. Although I think that feels more conversational and less precise compared to parentheses. Better for prose, not so much for technical documentation.

> If you don't care, why would you specify a parameter for it?

This extension method replaces the 5 separate extension methods (e.g. `builder.Creatable(true).Listable(false)`), but you don't always care about each. If I write `builder.SetAbilities(creatable: true, listable: false)` it says that I want to allow content item creation from the dashboard and I want to deny listing, but it says nothing about drafable becuase I don't care. It communicates that user requirement or software logic depends on the former two but not the latter. So that can be changed without consequence in the future.

Reply:
I'm ready to agree on **either** parentheses **or** `i.e.`. For me, `i.e.` is more natural (but I'm not a native speaker, so that only counts as an indication) and I don't feel what you feel:
> Better for prose, not so much for technical documentation.

So, just use either one of them.

As for the latter part of your answer, I now understand it much better.

Reply:
Interesting discussion. I frequently use `i.e., ...` (note the comma) because this is how Grammarly taught me. I use parentheses, too.

Reply:
Please link to the respective Grammarly article, because that comma seems wrong.

Reply:
I don't know when it changed, but now it accepts both versions (i.e. with and without comma). I clearly remember my surprise when Grammarly corrected me for the first time because of the missing comma. Weird.